lib/src/settings/embedder.dart

```diff
@@ -0,0 +1,262 @@
+import './distribution.dart';
+
+sealed class Embedder {
+  final String source;
+
+  const Embedder({
+    required this.source,
+  });
+
+  Map<String, Object?> toMap();
+
+  factory Embedder.fromMap(Map<String, Object?> map) {
+    final source = map['source'];
+
+    return switch (source) {
+      'openAi' => OpenAiEmbedder.fromMap(map),
+      'huggingFace' => HuggingFaceEmbedder.fromMap(map),
+      'userProvided' => UserProvidedEmbedder.fromMap(map),
+      'rest' => RestEmbedder.fromMap(map),
+      'ollama' => OllamaEmbedder.fromMap(map),
+      _ => throw Exception('unexpected embedder source'),
```

Comment:
This will be a breaking change if meilisearch adds new embedder versions, so I suggest adding an `UnkownEmbedder` class that stores the map as is.

Reply:
Yes, this makes more sense, however, I guess I'll need to update the structure a bit. It seems that I need to either remove source field in the base class or make it nullable as unknown embedder's source field might not be defined.

Reply:
I've updated the code, now we return unknown embedder with the original data when the source is not handled.